service/history/replication/stream_sender.go

```diff
@@ -489,10 +489,6 @@ Loop:
 			return fmt.Errorf("ReplicationServiceError StreamSender unable to get next replication task: %w", err)
 		}
 
-		if !s.shouldProcessTask(item) {
```

Comment:
line 562 below https://github.com/temporalio/temporal/pull/7061/files#diff-3a5383e58f4357884dc90cd45cdc6b2650d1bf8e59c6dbb2d78ca561af96da57R562 , the code will send `WorkflowReplicationMessages` with `endExclusiveWatermark`. That is not enough to update Ack level?